### PR TITLE
Update README, resolve some todo and add tests

### LINKED ISSUE #78: Refactor ctor DetectionDetail: not all codepage names are supported by .NET
Hello!

Refactor ctor DetectionDetail(string, float, CharsetProber, TimeSpan?, string): not all codepage names are supported by .NET.

Code like `Split('(').First().Trim()` looks redundant.

To do this, will need to consider that not all codepage names are supported by .NET (see `CodepageName` by #74). Also, it will be necessary to indicate explicitly in the `readme` that not all `Encoding` exists.

And maybe it will be possible to get rid of try / catch?

- [ ] Resolve __TODO__ in `CodepageName.cs`
  - [x] Resolve #75 
  - [x] Resolve #82 
  - [ ] Not supported _iso-2022-ch_? Maybe fix to _x-cp50227_?
  - [x] Not supported _CP949_? Attention!: it is superset of _ks_c_5601-1987_ (see https://github.com/CharsetDetector/UTF-unknown/pull/74#issuecomment-550362133)
  - [ ] `TODO: Not supported?` -- If possible, get rid of all such or process separately
  - [x] `TODO: Are other aliases ... ` (I think can simply remove `TODO:` because important identifier)
- [ ] \(Optional) to get rid of `try / catch` in ctor DetectionDetail
- [ ] Update `README.md`

### PR DESCRIPTION
**Pull request by rstm-sf**
Resolve  #78

What do you need to do for this?
- [x] Resolve `TODO: Not supported?` in `CodepageName.cs` -- If possible, get rid of all such or process separately
- [x] ~~\(Optional) to get rid of `try / catch` in ctor DetectionDetail~~
- [x] Update `README.md`

### CODE REVIEW

**Comment by rstm-sf** [comment]
See [SO](https://stackoverflow.com/q/18518730) for `X-ISO-10646-UCS-4-3412` and `X-ISO-10646-UCS-4-2143`

**Comment by rstm-sf** [comment]
> [VISCII](https://en.wikipedia.org/wiki/VISCII) is an unofficially-defined modified ASCII character encoding for using the Vietnamese language with computers.

**Comment by rstm-sf** [comment]
> euc-tw, iso-8859-10, iso-8859-16

Perhaps help answer in dotnet/corefx#42623

**Comment by rstm-sf** [comment]
> > euc-tw, iso-8859-10, iso-8859-16
> 
> Perhaps help answer in [dotnet/corefx#42623](https://github.com/dotnet/corefx/issues/42623)

It's not supported

**Comment by rstm-sf** [comment]
@304NotModified, hello!
 
>     * (Optional) to get rid of `try / catch` in ctor DetectionDetail

It looks like it will have to be left. Because, I was so glad that you can support other encodings in Core that I forgot about Linux :( And it seems that the user himself must write similar code before using the library
```c#
Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

**Comment by rstm-sf** [comment]
It looks like this works for Ubuntu too

**Comment by 304NotModified** [comment]
We could run the tests alsof on Travis CI (mono/net core) on Mac or Linux. 

What do you think?

**Comment by 304NotModified** [comment]
(maybe azure devops also supports Linux)

**Comment by rstm-sf** [comment]
> What do you think?

To be honest, I'm not good at it :) Perhaps it’s worth it, since it seems to me that the encodings are platform-dependent

I just thought that if definitions are available, are implementations on different OS available? Because it seemed to me that Codepage calls win32api and I got confused in the documentation

And the main question I’m thinking about now is whether we should add additional encodings, or should this be left to the discretion of the user? Which approach is better?

**Comment by rstm-sf** [comment]
> (maybe azure devops also supports Linux)

By the way, I recently left a [link](https://dev.to/itnext/the-ultimate-free-ci-cd-for-your-open-source-projects-3bkd) to bookmarks for review

**Comment by 304NotModified** [comment]
> And the main question I’m thinking about now is whether we should add additional encodings, or should this be left to the discretion of the user? Which approach is better?

I prefer the first :)

**Comment by rstm-sf** [comment]
> Add batch-file test for unsupported encoding

Also, it would be necessary to highlight tests for encodings associated with this dictionary
https://github.com/CharsetDetector/UTF-unknown/blob/303f0243024bd24fc1733dbcc47468ddf0eebf1e/src/DetectionDetail.cs#L18-L24

But this is another task

**Comment by rstm-sf** [comment]
>      (Optional) to get rid of `try / catch` in ctor DetectionDetail

This cannot be done because #100

**Comment by rstm-sf** [comment]
@304NotModified, what do you think, is it worth pointing out in README that the result may turn out different when detected based on bytes? https://github.com/CharsetDetector/UTF-unknown/issues/38#issuecomment-506960736

In my opinion, we need to determine one algorithm for all options, but this will change the result for some

**Comment by rstm-sf** [comment]
It seems that everything is ready, the following questions remained:



> because #100



> @304NotModified, what do you think, is it worth pointing out in README that the result may turn out different when detected based on bytes? [#38 (comment)](https://github.com/CharsetDetector/UTF-unknown/issues/38#issuecomment-506960736)

**Comment by rstm-sf** [comment]
> Build FAILED.

Ok.

**Comment by rstm-sf** [comment]
> It seems that everything is ready, the following questions remained:
> 
> > @304NotModified, what do you think, is it worth pointing out in README that the result may turn out different when detected based on bytes? [#38 (comment)](https://github.com/CharsetDetector/UTF-unknown/issues/38#issuecomment-506960736)

It’s probably better to consider it separately.

File: src/UTF-unknown.csproj [hunk 1]
```diff
@@ -17,7 +17,7 @@
   </PropertyGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' Or '$(TargetFramework)' == 'netstandard2.0' ">
-    <PackageReference Include="System.Text.Encoding.CodePages" Version="4.6.0" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="4.7.0" />
```
Comment (rstm-sf) [inline src/UTF-unknown.csproj]: It's normal?

Comment (304NotModified) [reply src/UTF-unknown.csproj]: What's normal?

Comment (rstm-sf) [reply src/UTF-unknown.csproj]: Changing the libraries we depend on is also a change to the API?

Comment (304NotModified) [reply src/UTF-unknown.csproj]: I don't think there is a problem updating this to 4.7.0

**Comment by 304NotModified** [comment]
Looks good to me, so merged :)

**Pull request merged**